### ISSUE TITLE
Cloudflare worker support broken since ~v7.15.0

### ISSUE THREAD

**Issue by nkohari** [workos/workos-node#1090]
Hey there! I'm getting these errors on startup when using WorkOS under wrangler 3.67.1 and node v22.4.0:

```
✘ [ERROR] Could not resolve "crypto"

    ../../node_modules/@peculiar/webcrypto/build/webcrypto.js:9:21:
      9 │ var crypto = require('crypto');
        ╵                      ~~~~~~~~

  The package "crypto" wasn't found on the file system but is built into node.
  Add "node_compat = true" to your wrangler.toml file and make sure to prefix the module name with "node:" to enable Node.js compatibility.


✘ [ERROR] Could not resolve "process"

    ../../node_modules/@peculiar/webcrypto/build/webcrypto.js:10:22:
      10 │ var process = require('process');
         ╵                       ~~~~~~~~~

  The package "process" wasn't found on the file system but is built into node.
  Add "node_compat = true" to your wrangler.toml file and make sure to prefix the module name with "node:" to enable Node.js compatibility.
```

Downgrading to `@workos-inc/node` version 7.14.0 fixes the problem.

Looking through recent commits, I think the addition of this dependency on `iron-session` broke things: https://github.com/workos/workos-node/compare/v7.14.0...v7.15.0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R40

**Comment by PaulAsjes** [comment]
Thanks for reporting! We had to use an older version of `iron-session` as we still need to support Node 16. We're going to look into abstracting the bits that we need from that library so we can continue to support both Node 15 and Cloudflare workers. 

Also going to look into adding a CFW test environment so we don't accidentally break CFW support in the future.

**Comment by PaulAsjes** [comment]
You'll likely still run into problems on v7.14.0. Downgrading to 6.8.0 should unblock you while we work on a fix.

**Comment by grinich** [comment]
@nkohari thanks for the report. Will have the fix for this shipped early next week ✌️

**Comment by nkohari** [comment]
Thanks! (And, hi Paul! 😄)

**Comment by MTG2000** [comment]
Not sure if this is the same issue, but we started getting this error on Vercel when we upgraded from 7.13.0 to 7.15.0

```
Error: [unenv] crypto.getCiphers is not implemented yet!
    at (index.mjs:1:412268)
    at (index.mjs:1:412562)
    at (index.mjs:1:470867)
    at (index.mjs:1:472050)
    at (index.mjs:1:475370)
```

**Comment by PaulAsjes** [comment]
Sup Nate 😎 

We have a [PR](https://github.com/workos/workos-node/pull/1091) up that should fix this (and the `Illegal invocation` error). We'll let you know when it's released!

**Comment by mattgd** [comment]
Hi @nkohari and @MTG2000 - thanks for the reports. We've just released [version 7.19.0](https://github.com/workos/workos-node/releases/tag/v7.19.0), which should fix worker/edge support. We also added an additional test to prevent this sort of issue from slipping through the cracks in the future.

Could you confirm this is working for you?

**Comment by nkohari** [comment]
Unfortunately, version 7.19.0 introduces a different problem when running under wrangler:

```
✘ [ERROR] ReferenceError: process is not defined

      at new WorkOS2
  (file:///Users/nkohari/Work/ardent/packages/api/node_modules/@workos-inc/node/lib/workos.js:60:87)
      at new WorkOSWorker
  (file:///Users/nkohari/Work/ardent/packages/api/node_modules/@workos-inc/node/lib/index.worker.js:35:1)
      at null.<anonymous>
```

Confirmed that I did correctly upgrade:

```
❯ npm ls @workos-inc/node
ardent-ai@ /.../ardent
└─┬ @ardent-ai/api@0.0.1 -> ./packages/api
  └── @workos-inc/node@7.19.0
```

**Comment by mattgd** [comment]
@nkohari Could you share your wrangler.toml with any sensitive strings removed?

**Comment by nkohari** [comment]
Sure:

```
name = "ardent-api"
compatibility_date = "2024-05-24"

[[d1_databases]]
binding = "DB"
database_name = "..."
database_id = "..."
migrations_dir = "drizzle"
```

**Comment by mattgd** [comment]
@nkohari Looks like it's caused by [this line](https://github.com/workos/workos-node/blob/main/src/workos.ts#L76).

As a temporary fix, you could provide your WorkOS client ID when instantiating the `WorkOS` client:
```
WorkOS('key', { clientId: '###' })
```
That being said looking into a permanent fix now.

**Comment by nkohari** [comment]
That fixed it, thanks! FWIW, it wouldn't be a big deal if the `clientId` was always required.

**Comment by mattgd** [comment]
Great, and appreciate that input. We're likely to move towards requiring `clientId` in a future major version. It'd also simplify some method calling in parts of the SDK.

Will have a permanent fix coming up shortly.

**Comment by mattgd** [comment]
Released a fix for the `process` ReferenceError bug as part of [v7.20.0](https://github.com/workos/workos-node/releases/tag/v7.20.0). Closing this issue out, but feel free to open another issue if you're still seeing errors.

**Comment by enmanuelr** [comment]
It looks like this issue was reintroduced in `v7.33.0`. I am experimenting build errors due to an unresolved crypto import with cloudflare workers. I was able to downgrade to `v7.32.0` and I can at least perform a successful build. Will report back if I see any runtime errors.

```
✘ [ERROR] Build failed with 1 error:

  ✘ [ERROR] Could not resolve "crypto"

      ../../node_modules/@workos-inc/node/lib/common/crypto/NodeCryptoProvider.js:36:36:
        36 │ const crypto = __importStar(require("crypto"));
           ╵                                     ~~~~~~~~

    The package "crypto" wasn't found on the file system but is built into node.
    - Make sure to prefix the module name with "node:" or update your compatibility_date to
  2024-09-23 or later.
```
